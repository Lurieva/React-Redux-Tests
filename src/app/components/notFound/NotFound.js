import React from 'react';

import './notFound.scss';

import { Button } from '../../shared';

const NotFound = () => (
    <div className="notFound-container">
        Page doesn`t Found
        <Button name='Back to Main Page'
                onClick={() => window.location = '/'}>
        </Button>
    </div> 
);

export default NotFound;