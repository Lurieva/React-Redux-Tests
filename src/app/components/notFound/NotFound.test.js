import React from 'react';
import { shallow } from 'enzyme';
import NotFound from './NotFound';

jest.mock('../../shared/Button/Button', () => 'Button');

describe('NotFound', () => {
    let mockFn = jest.fn();

    it('should be defined', () => {
        expect(NotFound).toBeDefined();
    });

    it('should render correctly', () => {
        const tree = shallow(
            <NotFound />
        );

        expect(tree).toMatchSnapshot();
    });
});