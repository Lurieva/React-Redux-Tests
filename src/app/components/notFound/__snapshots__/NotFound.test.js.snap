// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="notFound__content"
      >
        <div
          className="notFound__content__title"
        >
          Page doesn\`t Found
        </div>
        <Button
          className="notFound__content__backBtn"
          name="Back to Main Page"
          onClick={[Function]}
        />
      </div>,
      "className": "notFound",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="notFound__content__title"
          >
            Page doesn\`t Found
          </div>,
          <Button
            className="notFound__content__backBtn"
            name="Back to Main Page"
            onClick={[Function]}
          />,
        ],
        "className": "notFound__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Page doesn\`t Found",
            "className": "notFound__content__title",
          },
          "ref": null,
          "rendered": "Page doesn\`t Found",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "notFound__content__backBtn",
            "name": "Back to Main Page",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "Button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="notFound__content"
        >
          <div
            className="notFound__content__title"
          >
            Page doesn\`t Found
          </div>
          <Button
            className="notFound__content__backBtn"
            name="Back to Main Page"
            onClick={[Function]}
          />
        </div>,
        "className": "notFound",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="notFound__content__title"
            >
              Page doesn\`t Found
            </div>,
            <Button
              className="notFound__content__backBtn"
              name="Back to Main Page"
              onClick={[Function]}
            />,
          ],
          "className": "notFound__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Page doesn\`t Found",
              "className": "notFound__content__title",
            },
            "ref": null,
            "rendered": "Page doesn\`t Found",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "notFound__content__backBtn",
              "name": "Back to Main Page",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "Button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
